util/launcher/src/lib.rs

```diff
@@ -190,6 +190,8 @@ impl Launcher {
         &self,
         block_assembler_config: Option<BlockAssemblerConfig>,
     ) -> Result<(Shared, SharedPackage), ExitCode> {
+        observe_listen_port_occupancy(&self.args.config.network.listen_addresses)?;
```

Comment:
It should be done in [`NetworkService::new`](https://github.com/nervosnetwork/ckb/blob/78379cdb0a5476534c1dd701845acc08d4036b82/network/src/network.rs#L759-L769). It validates configurations inside.